File: src/main/java/com/example/debug/engine/model/CalculationRequest.java
package com.example.debug.engine.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import java.util.List;

/* loaded from: classes.dex */
public class CalculationRequest {

    private Configuration configuration;
    private List<Panel> panels;
    private List<Panel> stockPanels;
    private ClientInfo clientInfo;

    public Configuration getConfiguration() {
        return this.configuration;
    }

    public void setConfiguration(Configuration configuration) {
        this.configuration = configuration;
    }

    public List<Panel> getStockPanels() {
        return this.stockPanels;
    }

    public void setStockPanels(List<Panel> list) {
        this.stockPanels = list;
    }

    public List<Panel> getPanels() {
        return this.panels;
    }

    public void setPanels(List<Panel> list) {
        this.panels = list;
    }

    public String tilesToString() {
        StringBuilder sb = new StringBuilder();
        for (Panel panel : this.panels) {
            if (panel.getCount() > 0) {
                sb.append(" " + panel.toString());
            }
        }
        return sb.toString();
    }

    public String baseTilesToString() {
        StringBuilder sb = new StringBuilder();
        for (Panel panel : this.stockPanels) {
            if (panel.getCount() > 0) {
                sb.append(" " + panel.toString());
            }
        }
        return sb.toString();
    }
    
    public ClientInfo getClientInfo() {
        return this.clientInfo;
    }
    
    public void setClientInfo(ClientInfo clientInfo) {
        this.clientInfo = clientInfo;
    }

    public static class Panel {
        private int count;
        private Edge edge;
        private boolean enabled;
        private String height;
        private int id;
        private String label;
        private String material = TileDimensions.DEFAULT_MATERIAL;
        private int orientation;
        private String width;

        public int getId() {
            return this.id;
        }

        public void setId(int i) {
            this.id = i;
        }

        public String getWidth() {
            return this.width;
        }

        public void setWidth(String str) {
            this.width = str;
        }

        public String getHeight() {
            return this.height;
        }

        public void setHeight(String str) {
            this.height = str;
        }

        public int getCount() {
            return this.count;
        }

        public void setCount(int i) {
            this.count = i;
        }

        public String getMaterial() {
            return this.material;
        }

        public void setMaterial(String str) {
            if (str != null) {
                this.material = str;
            }
        }

        public boolean isEnabled() {
            return this.enabled;
        }

        public void setEnabled(boolean z) {
            this.enabled = z;
        }

        public int getOrientation() {
            return this.orientation;
        }

        public void setOrientation(int i) {
            this.orientation = i;
        }

        public String getLabel() {
            return this.label;
        }

        public void setLabel(String str) {
            this.label = str;
        }

        public Edge getEdge() {
            return this.edge;
        }

        public void setEdge(Edge edge) {
            this.edge = edge;
        }

        @JsonIgnore
        public boolean isValid() {
            String str;
            String str2;
            try {
                if (!this.enabled || this.count <= 0 || (str = this.width) == null || Double.parseDouble(str) <= 0.0d || (str2 = this.height) == null) {
                    return false;
                }
                return Double.parseDouble(str2) > 0.0d;
            } catch (Exception unused) {
                return false;
            }
        }

        public String toString() {
            StringBuilder sb = new StringBuilder("[");
            sb.append(this.width);
            sb.append("x");
            sb.append(this.height);
            sb.append("]*");
            sb.append(this.count);
            sb.append(this.enabled ? "" : "-disabled");
            return sb.toString();
        }

        public static class Edge {
            private String bottom;
            private String left;
            private String right;
            private String top;

            public String getTop() {
                return this.top;
            }

            public void setTop(String str) {
                this.top = str;
            }

            public String getLeft() {
                return this.left;
            }

            public void setLeft(String str) {
                this.left = str;
            }

            public String getBottom() {
                return this.bottom;
            }

            public void setBottom(String str) {
                this.bottom = str;
            }

            public String getRight() {
                return this.right;
            }

            public void setRight(String str) {
                this.right = str;
            }
        }
    }
}

File: src/features/engine/model/calculation_request.rs
use serde::{Deserialize, Serialize};

use crate::{
    constants::MaterialConstants,
    enums::orientation::Orientation,
    features::engine::model::{client_info::ClientInfo, configuration::Configuration, performance_thresholds::PerformanceThresholds},
    scaled_math::ScaledNumber,
};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CalculationRequest {
    pub configuration: Configuration,
    pub panels: Vec<Panel>,
    pub stock_panels: Vec<Panel>,
    pub client_info: ClientInfo,
    pub performance_thresholds: PerformanceThresholds,
}

impl Default for CalculationRequest {
    fn default() -> Self {
        Self {
            configuration: Configuration::default(),
            panels: Vec::new(),
            stock_panels: Vec::new(),
            client_info: ClientInfo::default(),
            performance_thresholds: PerformanceThresholds::default(),
        }
    }
}

impl CalculationRequest {
    pub fn tiles_to_string(&self) -> String {
        let mut result = String::new();
        for panel in &self.panels {
            if panel.count > 0 {
                result.push(' ');
                result.push_str(&panel.to_string());
            }
        }
        result
    }

    pub fn base_tiles_to_string(&self) -> String {
        let mut result = String::new();
        for panel in &self.stock_panels {
            if panel.count > 0 {
                result.push(' ');
                result.push_str(&panel.to_string());
            }
        }
        result
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Panel {
    pub id: u32,
    pub width: String,
    pub height: String,
    pub count: u32,
    pub material: String,
    pub enabled: bool,
    pub orientation: Orientation,
    pub label: String,
    pub edge: Option<Edge>,
}

impl Panel {
    pub fn new(id: u32, width: &str, height: &str, count: u32, label: &str) -> Self {
        Self {
            id,
            width: width.to_string(),
            height: height.to_string(),
            count,

            material: MaterialConstants::DEFAULT_MATERIAL.to_string(),

            enabled: false,
            orientation: Orientation::default(),
            label: label.to_string(),
            edge: None,
        }
    }

    pub fn set_material(&mut self, material: Option<String>) {
        if let Some(mat) = material {
            self.material = mat;
        }
    }

    pub fn is_valid(&self) -> bool {
        if !self.enabled || self.count <= 0 {
            return false;
        }

        let width_valid = self.width.parse::<f64>().unwrap_or(0.0) > 0.0;
        let height_valid = self.height.parse::<f64>().unwrap_or(0.0) > 0.0;

        width_valid && height_valid
    }
}

impl std::fmt::Display for Panel {
    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
        let disabled = if self.enabled { "" } else { "-disabled" };
        write!(
            f,
            "[{}x{}]*{}{}",
            self.width, self.height, self.count, disabled
        )
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Edge {
    pub top: Option<String>,
    pub left: Option<String>,
    pub bottom: Option<String>,
    pub right: Option<String>,
}

impl Edge {
    pub fn new() -> Self {
        Self {
            top: None,
            left: None,
            bottom: None,
            right: None,
        }
    }
}
